vm/src/stdlib/itertools.rs

```diff
@@ -73,6 +73,66 @@ impl PyItertoolsChain {
     }
 }
 
+#[pyclass(name = "compress")]
+#[derive(Debug)]
+struct PyItertoolsCompress {
+    data: PyObjectRef,
+    selector: PyObjectRef,
+}
+
+impl PyValue for PyItertoolsCompress {
+    fn class(vm: &VirtualMachine) -> PyClassRef {
+        vm.class("itertools", "compress")
+    }
+}
+
+#[pyimpl]
+impl PyItertoolsCompress {
+    #[pymethod(name = "__new__")]
+    #[allow(clippy::new_ret_no_self)]
+    fn new(
+        _cls: PyClassRef,
+        data: PyObjectRef,
+        selector: PyObjectRef,
+        vm: &VirtualMachine,
+    ) -> PyResult {
+        let data_iter = get_iter(vm, &data)?;
+        let selector_iter = get_iter(vm, &selector)?;
+
+        Ok(PyItertoolsCompress {
+            data: data_iter,
+            selector: selector_iter,
+        }
+        .into_ref(vm)
+        .into_object())
+    }
+
+    #[pymethod(name = "__next__")]
+    fn next(&self, vm: &VirtualMachine) -> PyResult {
+        let selector_iter = get_iter(vm, &self.selector)?;
+        let sel_obj = call_next(vm, &selector_iter)?;
+
+        let verdict = objbool::boolval(vm, sel_obj.clone())?;
+
+        let data_iter = get_iter(vm, &self.data)?;
+        let data_obj = call_next(vm, &data_iter);
+
+        if verdict {
+            match data_obj {
+                Ok(ok) => return Ok(ok),
+                Err(e) => return Err(e),
+            }
+        } else {
+            self.next(vm)
```

Comment:
Note that instead of recursion, you may want to prefer a loop here, to prevent hitting recursion limits.

Reply:
Sounds like a good idea. I will change it.

Reply:
@windelbouwman I made changes as suggested. Please have a look.

Reply:
This change looks good to me! I had one last comment, then I will merge this work.